http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcProviderMetadataTest.java

```diff
@@ -0,0 +1,368 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2024 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wildfly.security.http.oidc;
+
+import org.wildfly.security.jose.util.JsonSerialization;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test OIDC json config class to return of values.
+ */
+public class OidcProviderMetadataTest {
+    private static OidcProviderMetadata oidcProviderMetadata;
+    private static OidcProviderMetadata emptyOidcProviderMetadata;
+    private static OidcProviderMetadata withoutBooleansOidcProviderMetadata;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        // load the control data
+        ByteArrayInputStream is = new ByteArrayInputStream(providerMetaData.getBytes());
+        oidcProviderMetadata =  JsonSerialization.readValue(is, OidcProviderMetadata.class);
+        is.close();
+
+        // control data to check variable inits in OidcProviderMetadata
+        is = new ByteArrayInputStream(emptyProviderMetaData.getBytes());
+        emptyOidcProviderMetadata =  JsonSerialization.readValue(is, OidcProviderMetadata.class);
+        is.close();
+
+        is = new ByteArrayInputStream(withoutBooleansProviderMetaData.getBytes());
+        withoutBooleansOidcProviderMetadata =  JsonSerialization.readValue(is, OidcProviderMetadata.class);
+        is.close();
+    }
+
+    @Test
+    public void testIssuer() throws Exception {
+        assertTrue("http://localhost:8080/realms/myrealm".equals(oidcProviderMetadata.getIssuer()));
+        assertNull(emptyOidcProviderMetadata.getIssuer());
```

Comment:
Would it make sense to add an assertion here for withoutBooleansProviderMetaData? That way, we are testing the expected values for all three cases (control data, empty, control data modified to leave out some options) in each test case.

Reply:
done